core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -223,20 +236,36 @@ public Query termsQuery(List values, @Nullable QueryParseContext context) {
     }
 
     private final String type;
+    private final MappedFieldTypeReference joinFieldTypeRef;
+    private final MappedFieldType defaultJoinFieldType;
 
-    protected ParentFieldMapper(MappedFieldType fieldType, String type, Settings indexSettings) {
-        super(NAME, setupDocValues(indexSettings, fieldType), setupDocValues(indexSettings, Defaults.FIELD_TYPE), indexSettings);
+    protected ParentFieldMapper(MappedFieldType fieldType, MappedFieldType joinFieldType, String type, Settings indexSettings) {
+        super(NAME, fieldType, Defaults.FIELD_TYPE, indexSettings);
         this.type = type;
+        this.joinFieldTypeRef = new MappedFieldTypeReference(joinFieldType);
```

Comment:
I don't think we need a ref here? This field type is always associated directly with this mapper.